src/mint/v2/with-single-asset.ts

```diff
@@ -0,0 +1,84 @@
+import algosdk, {ALGORAND_MIN_TX_FEE} from "algosdk";
+import AlgodClient from "algosdk/dist/types/src/client/v2/algod/algod";
+
+import {MINT_APP_CALL_ARGUMENTS, V2_MINT_INNER_TXN_COUNT} from "../constants";
+import {CONTRACT_VERSION} from "../../contract/constants";
+import {SupportedNetwork} from "../../util/commonTypes";
+import {PoolInfo} from "../../util/pool/poolTypes";
+import {getValidatorAppID} from "../../validator";
+import {isAlgo} from "../../util/asset/assetUtils";
+
+export async function generateTxns({
+  client,
+  network,
+  poolAddress,
+  asset_1,
+  asset_2,
+  liquidityToken,
+  initiatorAddr
+}: {
+  client: AlgodClient;
+  pool: PoolInfo;
+  network: SupportedNetwork;
+  poolAddress: string;
+  asset_1: {id: number; amount: number | bigint};
+  asset_2: {id: number; amount: number | bigint};
```

Comment:
What do you think about changing these parameters like this:
```suggestion
  assetIn: {id: number; amount: number | bigint};
  pairAssetID: number;
```